public/app/listeners/loadDataset.js

```diff
@@ -24,13 +23,16 @@ const createNewDataset = ({ id, name, filepath }) => {
 };
 
 const loadDataset = (mainWindow, db) => async (event, args) => {
-  const { fileLocation } = args;
+  const { fileLocation, id } = args;
```

Comment:
I might be wrong, but in `LoadDataset.js` there is no `id` passed as argument. Maybe you can let the id be generated in `createNewDataset` and use the return value to create the `filepath` you need. Be careful because this function is also used in `executePythonAlgorithm.js`

Reply:
You are right. I missed its use in `executePythonAlgorithm.js` and messed up with the `id` logic... I'll make the changes the way you suggested.